src/main/java/org/spongepowered/api/command/CommandResult.java

```diff
@@ -24,60 +24,43 @@
  */
 package org.spongepowered.api.command;
 
-import java.util.Optional;
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.util.ResettableBuilder;
+
+import java.util.OptionalInt;
 
 import javax.annotation.Nullable;
 
 /**
  * Represents the result of a command in Sponge.
  */
-public class CommandResult {
-    private static final CommandResult EMPTY = builder().build();
-    private static final CommandResult SUCCESS = builder().successCount(1).build();
-    private final Optional<Integer> successCount;
-    private final Optional<Integer> affectedBlocks;
-    private final Optional<Integer> affectedEntities;
-    private final Optional<Integer> affectedItems;
-    private final Optional<Integer> queryResult;
+public interface CommandResult {
```

Comment:
That would be handy especially if the `SendCommandEvent` was split into `Pre` and `Post`. I may leave that for another PR, however.